pkg/endpoint/bpf.go

```diff
@@ -806,17 +806,14 @@ func (e *Endpoint) runPreCompilationSteps(regenContext *regenerationContext) (he
 		if err != nil {
 			return false, err
 		}
-		// Clean up map contents
-		e.getLogger().Debug("flushing old PolicyMap")
-		e.policyDebug(nil, "runPreCompilationSteps flushing old PolicyMap")
-		err = e.policyMap.DeleteAll()
+
+		// Synchronize the in-memory realized state with BPF map entries,
+		// so that any potential discrepancy between desired and realized
+		// state would be dealt with by the following e.syncPolicyMap.
+		e.realizedPolicy.PolicyMapState, err = e.dumpPolicyMapToMapState()
```

Comment:
This backport isn't applying cleanly. Looks like we are missing an intermediate commit backport. /cc @joestringer @pchaigno

Reply:
Looking at the existing code, isn't the conflict just because of the missing `e.initPolicyMapPressureMetric()` and following line? Those can just be omitted AFAIK.
https://github.com/cilium/cilium/blob/748da8d42d9c85db1d9280b1bfeba1b4edaead50/pkg/endpoint/bpf.go#L804-L820

Reply:
>#12 105.6 ../pkg/endpoint/bpf.go:813:43: e.dumpPolicyMapToMapState undefined (type *Endpoint has no field or method dumpPolicyMapToMapState)
#12 122.7 make[1]: *** [Makefile:15: cilium-agent] Error 2

Looks like `dumpPolicyMapToMapState` was introduced in v1.10 (commit 9dc1350fdb81eca6dcf00076dd6ec39258e80d0a which wasn't backported).

Reply:
Ah, thanks! Totally missed that :facepalm: 

Maybe a simple call to `e.policyMap.DumpKeysToSlice()` instead would work? Or maybe we can copy only `dumpPolicyMapToMapState` to v1.9? The commit seems to have a lot of other changes with potential side effects. @jrajahalme probably knows best.

Reply:
I've dropped the commit for now.

Reply:
Copied the `dumpPolicyMapToMapState` func as suggested at https://github.com/cilium/cilium/pull/17390.